### PR TITLE
Update lock ttl when retry to pessimistic lock keys

### PR DESCRIPTION
**Pull request by longfangsong**
fix [tidb#31203](https://github.com/pingcap/tidb/issues/31203)

Signed-off-by: longfangsong <longfangsong@icloud.com>

### CODE REVIEW

**Comment by cfzjywxk** [comment]
@longfangsong 
We could continue finish the pr to finish the optimization.

File: integration_tests/2pc_test.go [hunk 1]
```diff
@@ -1345,6 +1345,55 @@ func (s *testCommitterSuite) TestAsyncCommit() {
 	})
 }
 
+func (s *testCommitterSuite) TestRetryPushTTL() {
```
Comment (longfangsong) [inline integration_tests/2pc_test.go:1348]: The same test case will fail [on master](https://github.com/tikv/client-go/compare/master...longfangsong:c13220036c3c9fa61097ddc7d93f10b449461c4f).

File: txnkv/transaction/pessimistic.go [hunk 2]
```diff
@@ -134,6 +134,9 @@ func (action actionPessimisticLock) handleSingleBatch(c *twoPhaseCommitter, bo *
 				req.PessimisticLock().WaitTimeout = timeLeft
 			}
 		}
+		elapsed := uint64(time.Since(c.txn.startTime) / time.Millisecond)
+		ttl := elapsed + atomic.LoadUint64(&ManagedLockTTL)
```
Comment (sticnarf) [inline txnkv/transaction/pessimistic.go:138]: The `ttl` calculated outside the loop looks useless now.

File: txnkv/transaction/pessimistic.go [hunk 3]
```diff
@@ -96,23 +95,11 @@ func (action actionPessimisticLock) handleSingleBatch(c *twoPhaseCommitter, bo *
 		}
 		mutations[i] = mut
 	}
-	elapsed := uint64(time.Since(c.txn.startTime) / time.Millisecond)
-	ttl := elapsed + atomic.LoadUint64(&ManagedLockTTL)
-	if _, err := util.EvalFailpoint("shortPessimisticLockTTL"); err == nil {
```
Comment (sticnarf) [inline txnkv/transaction/pessimistic.go:101]: I think we'd better keep the failpoint and move it into the loop. Some tests may depend on the failpoint.

**Review by sticnarf — APPROVED** [review]
LGTM

**Review by youjiali1995 — APPROVED** [review]

**Pull request merged**